src/components/Patient/allergy/AllergyTable.tsx

```diff
@@ -1,150 +1,262 @@
-                : ""
-            }`}
+    <div className="border-1 shadow rounded-md p-4 bg-white">
+      <div className="flex justify-between items-start flex-wrap gap-2">
+        <div className="flex-1 min-w-0">
+          <div className="text-base font-semibold text-gray-900 break-words">
+            {allergy.code.display}
+          </div>
+          <div className="italic text-gray-500">{t(allergy.category)}</div>
+        </div>
+
+        <div className="flex items-center gap-2">
+          <Badge
+            variant={ALLERGY_CLINICAL_STATUS_COLORS[allergy.clinical_status]}
+            className="whitespace-nowrap"
           >
-            <TableCell className="first:rounded-l-md">
-              <div className="flex items-center">
-                {CATEGORY_ICONS[allergy.category ?? ""]}
-              </div>
-            </TableCell>
-            <TableCell className="font-medium pl-0 md:whitespace-normal">
-              {allergy.code.display}
-            </TableCell>
-            <TableCell>
-              <Badge
-                variant={
-                  ALLERGY_CLINICAL_STATUS_COLORS[allergy.clinical_status]
-                }
-                className="whitespace-nowrap"
+            {t(allergy.clinical_status)}
+          </Badge>
+          <DropdownMenu>
+            <DropdownMenuTrigger asChild>
+              <Button
+                variant="link"
+                className="text-gray-500 hover:text-gray-700 p-1"
               >
-                {t(allergy.clinical_status)}
-              </Badge>
-            </TableCell>
-            <TableCell>
-              <Badge
-                variant={ALLERGY_CRITICALITY_COLORS[allergy.criticality]}
-                className="whitespace-nowrap"
-              >
-                {t(allergy.criticality)}
-              </Badge>
-            </TableCell>
-            <TableCell>
-              <Badge
-                variant={
-                  ALLERGY_VERIFICATION_STATUS_COLORS[
-                    allergy.verification_status
-                  ]
+                <BadgeInfo size={16} />
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent>
+              {allergy.note && (
+                <DropdownMenuItem
+                  onClick={() => setShowNote(!showNote)}
+                  className="flex items-center gap-2 px-3 py-2 font-semibold"
+                >
+                  <File className="size-4" />
+                  <span>{showNote ? t("hide_note") : t("see_note")}</span>
+                </DropdownMenuItem>
+              )}
+
+              <DropdownMenuItem
+                onClick={() =>
+                  navigate(
+                    facilityId
+                      ? `/facility/${facilityId}/patient/${patientId}/encounter/${allergy.encounter}/updates`
+                      : `/organization/organizationId/patient/${patientId}/encounter/${allergy.encounter}/updates`,
+                  )
```

Comment:
make this happen through a callback so that we don't need to pass facility and patient id as props